nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -330,15 +330,19 @@ impl DataStore {
     ) -> ListResultVec<IpPoolResource> {
         use db::schema::ip_pool;
         use db::schema::ip_pool_resource;
+        use db::schema::silo;
 
         paginated(
             ip_pool_resource::table,
             ip_pool_resource::ip_pool_id,
             pagparams,
         )
         .inner_join(ip_pool::table)
+        .inner_join(silo::table.on(silo::id.eq(ip_pool_resource::resource_id)))
         .filter(ip_pool::id.eq(authz_pool.id()))
         .filter(ip_pool::time_deleted.is_null())
+        .filter(silo::time_deleted.is_null())
+        .filter(silo::discoverable.eq(true))
```

Comment:
Matching the logic in `silos_list`. Since we're not joining on silo, we can also filter out links to deleted silos, even though it should be impossible to have any such links after #4867.

https://github.com/oxidecomputer/omicron/blob/0531d6161ee2b670a0ab2a72f26727d4ec3cacaf/nexus/db-queries/src/db/datastore/silo.rs#L322-L351